### PR TITLE
fix: validate JWT token on alias look ahead

### PR DESCRIPTION
**Pull request by benashz**
This PR adds support for validating that a request's JWT token contains the required claims for the bound role.  

# Overview
A high level description of the contribution, including:
Who the change affects or is for (stakeholders)?
What is the change? 
Why is the change needed?
How does this change affect the user experience (if at all)?

# Design of Change
Validate the JWT token against a known Vault role. If it is deemed to be invalid return an error.

Note: the token is not passed to Kubernetes token review API, as this is only needed for login.

### CODE REVIEW

File: scripts/update_deps.sh [hunk 1]
```diff
@@ -1,37 +0,0 @@
-#!/bin/sh
```
Comment (calvn) [inline scripts/update_deps.sh:1]: I wonder if the script changes should be a separate PR to make the backport/cherry-picking process easier.

Comment (benashz) [reply scripts/update_deps.sh:1]: Good point :)

Comment (benashz) [reply scripts/update_deps.sh]: Fixed.

**Review by tvoran — APPROVED** [review]

File: path_login.go [hunk 2]
```diff
@@ -190,15 +187,12 @@ func (b *kubeAuthBackend) aliasLookahead(ctx context.Context, req *logical.Reque
...
-	parsedJWT, err := jws.ParseJWT([]byte(jwtStr))
+	config, err := b.loadConfig(ctx, req.Storage)
 	if err != nil {
 		return nil, err
 	}
 
-	// Decode claims into a service account object
-	sa := &serviceAccount{}
-	err = mapstructure.Decode(parsedJWT.Claims(), sa)
+	sa, err := b.parseAndValidateJWT(jwtStr, role, config)
```
Comment (tomhjp) [inline path_login.go:195]: Is it worth leaving a comment behind to ensure the intent is clear and help prevent similar bugs in future patches? It seems like an easy mistake to make.

**Review by tomhjp — APPROVED** [review]

File: path_login.go (hunk 2 above)
Comment (benashz) [reply path_login.go:198]: Fixed in. https://github.com/hashicorp/vault-plugin-auth-kubernetes/pull/114/commits/aec50977ce5b5f1aab03c38c4153abec583105fe

**Review by tomhjp — APPROVED** [review]
